moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1062,24 +1062,29 @@ void planning_scene::PlanningScene::loadGeometryFromStream(std::istream &in, con
 
 void planning_scene::PlanningScene::setCurrentState(const moveit_msgs::RobotState &state)
 {
+  moveit_msgs::RobotState state_no_attached(state);
+  state_no_attached.attached_collision_objects.clear();
```

Comment:
Please add a comment above this line that explains why you do not pass on `attached_collision_objects` on to `robotStateMsgToRobotState`.
something along the lines of

> attached bodies will be processed separately by `processAttachedCollisionObjectMsg` after the state has been updated